aio.core/aio/core/pipe/abstract/pipe.py

```diff
@@ -0,0 +1,90 @@
+
+import asyncio
+import importlib
+import sys
+from functools import cached_property
+
+import abstracts
+
+from aio.core.functional import async_property
+
+
+class AStdinStdoutProcessor(metaclass=abstracts.Abstraction):
+
+    def __init__(self, stdin, stdout, processor, log=None):
+        self.processor = processor
+        self.stdin = stdin
+        self.stdout = stdout
+        self._log = log
+
+    @cached_property
+    def connecting(self):
+        return asyncio.Lock()
+
+    @async_property(cache=True)
+    async def connection(self):
+        reader = asyncio.StreamReader()
+        protocol = asyncio.StreamReaderProtocol(reader)
+        await self.loop.connect_read_pipe(lambda: protocol, self.stdin)
+        w_transport, w_protocol = await self.loop.connect_write_pipe(asyncio.streams.FlowControlMixin, self.stdout)
+        writer = asyncio.StreamWriter(w_transport, w_protocol, reader, self.loop)
+        return reader, writer
+
+    @async_property
+    async def listener(self):
+        async with self.connecting:
+            reader = await self.reader
+        self.log(f"LISTENING\n")
+        while True:
+            line = await reader.readline()
+            if not line:
+                self.log(f"CLOSED: {line}\n")
+                break
+            if not line.strip():
+                continue
+            # self.log(f"RCVD: {line}\n")
```

Comment:
remove?

Reply:
probs yes - i need to do a bit more work on this - i should probs mark it wip for now